### PR TITLE
Internal: Generic warning message for upload JSON [ED-11138]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: assets/dev/js/admin/admin.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import environment from '../../../../core/common/assets/js/utils/environment';
 import Events from 'elementor-utils/events';
 import FilesUploadHandler from '../editor/utils/files-upload-handler';
 import TemplateControls from './new-template/template-controls.js';
+import { showJsonWarningMessageIfNeeded } from '../utils/upload-json-warning-message';
```
Comment (StyleShit) [inline assets/dev/js/admin/admin.js:7]:
```suggestion
import { showJsonUploadWarningMessageIfNeeded } from '../utils/json-upload-warning-message';
```

maybe

File: assets/dev/js/editor/components/template-library/views/parts/import.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import FilesUploadHandler from '../../../../utils/files-upload-handler';
+import { showJsonWarningMessageIfNeeded } from '../../../../../utils/upload-json-warning-message';
```
Comment (StyleShit) [inline assets/dev/js/editor/components/template-library/views/parts/import.js:2]:
```suggestion
import { showJsonWarningMessageIfNeeded } from 'elementor-utils/upload-json-warning-message';
```

File: assets/dev/js/utils/introduction.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
 export default class extends elementorModules.Module {
+	introductionMap = null;
+
 	constructor( ...args ) {
 		super( ...args );
 
 		this.initDialog();
 	}
 
+	setIntroductionMap( source ) {
```
Comment (StyleShit) [inline assets/dev/js/utils/introduction.js:10]:
```suggestion
	setIntroductionMap( map ) {
```

OR 

```suggestion
	setIntroductionMap( introductionMap ) {
```

IDK

File: assets/dev/js/utils/upload-json-warning-message.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+const genericMessageIntroductionKey = 'upload_json_warning_generic_message';
+/**
+ * @type {import('../utils/introduction').default | null}
+ */
+let genericWarningModal = null;
+
+export function showJsonWarningMessageIfNeeded( { introductionMap, Introduction, waitForSetViewed = false } ) {
+	if ( ! genericWarningModal ) {
+		genericWarningModal = createGenericWarningModal( Introduction );
```
Comment (StyleShit) [inline assets/dev/js/utils/upload-json-warning-message.js:9]:
```suggestion
		genericWarningModal = createGenericWarningModal( IntroductionClass );
```

maybe?

File: assets/dev/js/utils/upload-json-warning-message.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+function createGenericWarningModal( Introduction ) {
+	const dialogId = 'e-generic-warning-modal-for-json-upload';
+
+	const introduction = new Introduction( {
+		introductionKey: genericMessageIntroductionKey,
+		dialogType: 'confirm',
+		dialogOptions: {
+			id: dialogId,
+			headerMessage: __( 'Warning: JSON files may be unsafe', 'elementor' ),
+			// Translators: %1$s: Opening strong tag, %2$s: Closing strong tag.
```
Comment (StyleShit) [inline assets/dev/js/utils/upload-json-warning-message.js:64]:
```suggestion
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/admin/admin.js (hunk 1 above)
Comment (Nevoss) [reply assets/dev/js/admin/admin.js:7]: Done

File: assets/dev/js/editor/components/template-library/views/parts/import.js (hunk 2 above)
Comment (Nevoss) [reply assets/dev/js/editor/components/template-library/views/parts/import.js:2]: Done

File: assets/dev/js/utils/introduction.js (hunk 3 above)
Comment (Nevoss) [reply assets/dev/js/utils/introduction.js:10]: Done

File: assets/dev/js/utils/upload-json-warning-message.js (hunk 5 above)
Comment (Nevoss) [reply assets/dev/js/utils/upload-json-warning-message.js:64]: Opss.. done

**Review by StyleShit — DISMISSED** [review]

**Review by StyleShit — APPROVED** [review]

**Pull request merged**